atomic_reactor/plugins/pre_bump_release.py

```diff
@@ -109,37 +109,40 @@ def run(self):
 
         parser = df_parser(self.workflow.builder.df_path, workflow=self.workflow)
         dockerfile_labels = parser.labels
+        labels = Labels(dockerfile_labels)
 
-        release = get_preferred_label(dockerfile_labels, 'release')
-        if release is not None and not self.append:
-            self.log.debug("release set explicitly so not incrementing")
-            return
+        try:
+            release_val = labels.get_name_and_value(Labels.LABEL_TYPE_RELEASE)[1]
+            if not self.append:
+                self.log.debug("release set explicitly so not incrementing")
+                return
+        except:
+            release_val = None
+
+        release = labels.get_name(Labels.LABEL_TYPE_RELEASE)
+        component_label = labels.get_name(Labels.LABEL_TYPE_COMPONENT)
 
-        component_label = get_preferred_label_key(dockerfile_labels,
-                                                  'com.redhat.component')
         try:
             component = dockerfile_labels[component_label]
         except KeyError:
             raise RuntimeError("missing label: {}".format(component_label))
 
-        version_label = get_preferred_label_key(dockerfile_labels, 'version')
+        version_label = labels.get_name(Labels.LABEL_TYPE_VERSION)
         try:
             version = dockerfile_labels[version_label]
         except KeyError:
             raise RuntimeError('missing label: {}'.format(version_label))
 
         if self.append:
-            next_release = self.get_next_release_append(component, version, release)
+            next_release = self.get_next_release_append(component, version, release_val)
         else:
             next_release = self.get_next_release_standard(component, version)
 
         # Always set preferred release label - other will be set if old-style
         # label is present
-        release_labels = get_all_label_keys('release')
-        preferred_release_label = get_preferred_label_key(dockerfile_labels,
-                                                          'release')
-        old_style_label = get_all_label_keys('com.redhat.component')[1]
-        release_labels_to_be_set = [preferred_release_label]
+        release_labels = labels.LABEL_NAMES[Labels.LABEL_TYPE_RELEASE]
+        old_style_label = labels.LABEL_NAMES[Labels.LABEL_TYPE_COMPONENT][1]
+        release_labels_to_be_set = [release]
```

Comment:
delete this, until and including for loop

and leave only that
self.log.info
AND
dockerfile_labels[release_label] = next_release

Reply:
The entire point of the PR is to be able to set an unspecified, but more than two, number of release labels. So removing the ability to set more than one release label seems like a bad solution.

Reply:
Fixed to match the updated JIRA task.